### PR TITLE
Running cargo-features-manager prune.

### PR DESCRIPTION
**Pull request by dessalines**
- Context #5273

Used https://github.com/ToBinio/cargo-features-manager to prune some unneeded features.

### CODE REVIEW

**Review by SleeplessOne1917 — APPROVED** [review]

**Comment by Nutomic** [comment]
This PR doesnt remove any feature, just reorders them.

**Comment by Nothing4You** [comment]
it removes `serde-compat` from `ts-rs` in the workspace and `serde` from `uuid`, as well as `tokio1` from `lettre` in the `utils` crate. everything else is just reordering.
for `lettre`, `tokio1-rustls-tls` implies `tokio1`, so there is no difference there, but the other two seem to actually be different.
i'm not sure about the details of workspace dependencies though, the workspace `uuid` dependency specifies the `serde` feature, the `utils` crate specifies the `v4` feature but no longer the `serde` feature. other crates don't explicitly specify `uuid` features.

File: crates/utils/Cargo.toml [hunk 1]
```diff
@@ -77,9 +77,8 @@ deser-hjson = { version = "2.2.4", optional = true }
 smart-default = { version = "0.7.1", optional = true }
 lettre = { version = "0.11.10", default-features = false, features = [
   "builder",
-  "tokio1",
```
Comment (dessalines) [inline crates/utils/Cargo.toml:80]: @Nutomic it did find one here, and also on L71 removing serde.

It took like an hour to run, and also created a `target` dir of > 100GB lol, but at least it did find a few things.

**Comment by dessalines** [comment]
As always with rust, we'd have compilation / CI failures if it broke anything.

File: crates/utils/Cargo.toml (hunk 1 above)
Comment (Nutomic) [reply crates/utils/Cargo.toml:80]: Better than nothing, but this definitely wont have any noticable effect on build time.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**